### PR TITLE
Fix golang for vendor hash changes

### LINKED ISSUE #203: buildGoModule: modSha256 -> vendorSha256
In https://github.com/NixOS/nixpkgs/pull/86376 we renamed modSha256 to vendorSha256. No new packages with modSha256 will be accepted hence you can just rename the attribute to the new version.

### PR DESCRIPTION
**Pull request by bhipple**
Resolves #203

CC @mic92 @c00w @ryantm

### CODE REVIEW

File: src/Rewrite.hs [hunk 1]
```diff
@@ -185,27 +185,32 @@ rustCrateVersion log args@(Args _ attrPth drvFile drvContents) = do
...
+      lift . log $ "Found old vendorSha256 = " <> oldVendorSha256
+      -- Sometimes we just use the srcs vendored in the repo
+      if (oldVendorSha256 == "null")
+        then do
+          lift . log $ "vendorSha256 is explicitly set to null; leaving it to use repo vendored src"
+        else do
+          _ <- lift $ File.replaceIO oldVendorSha256 Nix.sha256Zero drvFile
+          newVendorSha256 <- Nix.getHashFromBuild attrPth
+          _ <- lift $ File.replaceIO Nix.sha256Zero newVendorSha256 drvFile
+          -- Note that on some small bumps, this may not actually change if go.sum did not
```
Comment (bhipple) [inline src/Rewrite.hs:209]: For example:
https://github.com/NixOS/nixpkgs/pull/89683

**Comment by bhipple** [comment]
@ryantm the bot has lost some understanding of what go packages in NixPkgs look like and is sending uninformed PRs.  I think these should fail safely at build time, but we should deploy this with some urgency.

I've tested the 3 cases above.

**Comment by ryantm** [comment]
This is deployed now; thanks!

**Comment by c00w** [comment]
One thing that might be awesome, is if this code could try vendor=null, then fallback to manually setting the hash. I'll go modify my code to see if it can do a vendor=null try across the repo.

**Comment by c00w** [comment]
I'm guessing this code expects one build per hash? (I guess another corner case is the old version had vendor=null, but the new version fails with it).

**Comment by bhipple** [comment]
> One thing that might be awesome, is if this code could try vendor=null, then fallback to manually setting the hash. 

This is easily do-able by modifying the logic here:
https://github.com/ryantm/nixpkgs-update/blob/master/src/Rewrite.hs#L202-L213

Ideally you could give it a shot and send a PR, but either way feel free to open an issue and I'll give it a try someday.

**Comment by c00w** [comment]
I'll take a hack - looks like I can chain WithExceptT calls to try null, then non null.

**Comment by c00w** [comment]
Done - https://github.com/ryantm/nixpkgs-update/pull/207

**Pull request merged**